src/v/raft/configuration_manager.cc

```diff
@@ -367,11 +367,20 @@ configuration_manager::start(bool reset, model::revision_id initial_revision) {
     });
 }
 
-ss::future<> configuration_manager::maybe_store_highest_known_offset(
-  model::offset offset, size_t bytes) {
+void configuration_manager::maybe_store_highest_known_offset_in_background(
+  model::offset offset, size_t bytes, ss::gate& gate) {
     _highest_known_offset = offset;
-    _bytes_since_last_offset_update += bytes;
 
+    ssx::spawn_with_gate(gate, [this, bytes] {
+        return do_maybe_store_highest_known_offset(bytes);
+    });
+}
+
+ss::future<>
+configuration_manager::do_maybe_store_highest_known_offset(size_t bytes) {
+    auto lock = co_await _hko_checkpoint_lock.get_units();
```

Comment:
just had a thought - maybe this mutex could simply be a boolean flag, waiting on it doesn't seem to be worth it. WDYT?

Reply:
it may be helpful if one of the fibers fail, but other than that yeah, seems like a boolean would be enough.

Reply:
I guess if one of the fibers fails, other are likely to fail as well so it doesn't make much sense to retry on such a short notice.